File: FitTrackAPI/Models/Exercise.cs
public class Exercise
{
	public int Id { get; set; }
	public string Name { get; set; }
	public string Image { get; set; }
	public string Description { get; set; }

	public ICollection<Set> Sets { get; set; }
	public ICollection<ExerciseInTraining> ExercisesInTraining { get; set; }
}

File: fittrack_mobile_app/lib/models/exercise.dart
import 'set.dart';


class Exercise {
  final int id;
  final String? name;
  final String? description;
  final List<Set>? sets;

  Exercise({
    required this.id,
    this.name,
    this.description,
    this.sets,
  });

  // Factory constructor to create an Exercise instance from a JSON object
  factory Exercise.fromJson(Map<String, dynamic> json) {
    return Exercise(
      id: json['id'] as int,
      name: json['name'] as String?,
      description: json['description'] as String?,
      sets: (json['sets'] as List<dynamic>?)
          ?.map((set) => Set.fromJson(set as Map<String, dynamic>))
          .toList(),
    );
  }

  // Method to convert an Exercise instance into a JSON object
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'sets': sets?.map((set) => set.toJson()).toList(),
    };
  }
}
